### DISCUSSION TITLE
zigbee quiz

### DISCUSSION THREAD

**Discussion by ffgomez** [arendst/Tasmota#22558] [Q&A] [answered]
Dear Community,

I’m planning to update my Tasmota infra by integrating Zigbee, so I ordered a bunch of zigbee switches and esp32-c6. My idea is to maintain the same concept as before, but now the Tasmota devices could also be triggered by low-energy sensors and switches. However, I been researching and this is not common by any mean, so any guideance will be higly appreciated as it will save me some precious hours, I am willing to research and build but not to develop something brand new if it doesn't exist

Questions:
- Does this make sense?
From my perspective, this clustered approach seems to combine the best of both worlds. However, most configurations I’ve seen separate Zigbee and Ethernet networks, connected through a single dedicated zigbee to ethernet router (zigbee2tasmota=dedicated or zigbee2mqtt=dockerized ). What I’m proposing is a hybrid coordinator/endpoint WiFi Tasmota device. Is this a good approach?

- ESP32-C6 limitations:
I understand that the ESP32-C6 can either use WiFi or Zigbee, but not both simultaneously. This means I would need to add a UART Zigbee module and flash it, correct?
For reference:
https://tasmota.github.io/docs/CC2530/
https://es.aliexpress.com/item/1005007991154254.html -> just wired, no more elements

- Compiling Tasmota-Zigbee for ESP32-C6:
From what I’ve seen, I’ll need to compile my own Tasmota-Zigbee (zigbee2tasmota perhaps?) image for the ESP32-C6, as there doesn’t seem to be a prebuilt version available. Are there any known issues or challenges with this? Or, is there any magical wizard in PlatformIO in VSCode that will allow me to select Tasmota-Zigbee and ESP32-C6 and press "compile"? I really haven't found anything this close in the vscode addon

- Adding Zigbee endpoints:
Once flashed, will I be able to use my 4th and 5th pin for Zigbee UART? and then, will Tasmota provide an easy way (e.g., buttons or a web interface) to add Zigbee endpoints and manage networks as I've seen in zigbee2tasmota tutorials?
https://templates.blakadder.com/ESP32-C6-DevKitC.html

- Scripting capabilities:
Will I be able to create scripts within Tasmota to, for example, send an MQTT trap when a Zigbee reed sensor is triggered or turn on a tasmota's pin LED when an IKEA switch is pressed? Could be also that all zigbee traffic is forwarded to mqtt?

- Zigbee topology:
Shall I create the same zigbee network or different networks per tasmota?

- Bonus question:
Is it possible to compile a Tasmota-Zigbee image that also allows flashing CC2530 modules? The idea would be to have one image to rule them all. 😊

Cheers!

**Comment by s-hadinger** [comment, marked as answer]
> * From my perspective, this clustered approach seems to combine the best of both worlds. However, most configurations I’ve seen separate Zigbee and Ethernet networks, connected through a single dedicated zigbee to ethernet router (zigbee2tasmota=dedicated or zigbee2mqtt=dockerized ). What I’m proposing is a hybrid coordinator/endpoint WiFi Tasmota device. Is this a good approach?

I'm not sure to understand what you mean with clustered. Fragmenting your Zigbee network is not necessarily a good idea when it comes to radio spectrum, you may have overlapping zigbee networks on the same channels, and this is not good.



> * I understand that the ESP32-C6 can either use WiFi or Zigbee, but not both simultaneously. This means I would need to add a UART Zigbee module and flash it, correct?

Correct, you always need 2 chips, one for Wifi and one for Zigbee. We generally recommend Ethernet to avoid cross pollution of Zigbee and Wifi.

> * Once flashed, will I be able to use my 4th and 5th pin for Zigbee UART? and then, will Tasmota provide an easy way (e.g., buttons or a web interface) to add Zigbee endpoints and manage networks as I've seen in zigbee2tasmota tutorials?

Not sure what you mean here. Tasmota does not support creating Zigbee end-devices; if this is what you mean.



> * Will I be able to create scripts within Tasmota to, for example, send an MQTT trap when a Zigbee reed sensor is triggered or turn on a tasmota's pin LED when an IKEA switch is pressed? Could be also that all zigbee traffic is forwarded to mqtt?
> *

Yes. Tasmota anyways publishes everything it receives to MQTT.



> * Shall I create the same zigbee network or different networks per tasmota?
> *

Again, not sure what you want to do. Zigbee networks accept only 1 coordinator. You cannot start 2 zigbee coordinators on the same network, the second will refuse to start.

**Comment by ffgomez** [comment]
Thanks a lot for your enlightment. It makes everything clear now. The RF spectrum issue you pointed out makes a lot of sense and really defines de infra. 

My premise was to update my current tasmota infra with low power switches/sensors, everything else was a pipe dream. I understood that the simpliest step was to add zigbee "capabilities" to my tasmotized and home integrated esp32s. So each of the tasmotized esp32s will have its "somwhow" own zigbee network.

**Reply by s-hadinger** [reply]
Keep it simple to a single Zigbee network in your home, managed by a single coordinator. Any other scheme will turn to be a can of worms, and a nightmare to manage

**Reply by sfromis** [reply]
When it comes to "simplest", I'd say that using Tasmota with Wifi is a preferable to the more complex usage of Zigbee.
